service/controller/v3/arm_templates/virtual_network_gateway.json

```diff
@@ -0,0 +1,89 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterID": {
+      "type": "string"
+    },
+    "GiantSwarmTags": {
+      "type": "object",
+      "defaultValue": {
+        "provider": "F80D01C0-7AAC-4440-98F6-5061511962AD"
+      }
+    },
+    "virtualNetworkName": {
+      "type": "string"
+    },
+    "vpnGatewayName": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "publicIPName": "[concat(parameters('clusterID'), '-VPNGateway-PublicIP')]",
+    "publicIPID": "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPName'))]",
+
+    "vnetGatewaySubnetName": "GatewaySubnet",
```

Comment:
We rely on this name in two places. I think we should make it a variable in main.json and pass it to both virtual_network_gateway.json and virtual_network_setup.json. Then we are DRY.

Reply:
done